app/src/main/java/id_iot/demo/PagerAdapter.java

```diff
@@ -0,0 +1,35 @@
+package id_iot.demo;
+
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentStatePagerAdapter;
+
+public class PagerAdapter extends FragmentStatePagerAdapter{
+    int mNumOfTabs;
+
+    public PagerAdapter(FragmentManager fragmentManager, int NumOfTabs) {
+        super(fragmentManager);
+        this.mNumOfTabs = NumOfTabs;
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        switch (position) {
+            case 0:
```

Comment:
I would suggest extracting constants. it is just easier to read switch(value) {
case MAIN_LIST:
break;

case SELECTED_RECIPE:
}

etc